workers/unity/Assets/Fps/Scripts/GameLogic/VisibilityAndCollision.cs

```diff
@@ -22,60 +21,36 @@ private void Awake()
 
         private void OnEnable()
         {
-            health.OnRespawn += OnRespawn;
-            health.OnHealthModified += OnHealthModified;
-            ApplyState();
-            SetIsDeadState(health.Data.Health <= 0);
+            health.HealthUpdated += HealthUpdated;
+            SetIsVisible(health.Data.Health > 0);
```

Comment:
This doesn't need to pass in a parameter as the only two times its called is with `health.Data.Health > 0` as the argument - can just move that into the function

Reply:
And then rename to `UpdateVisibility`

Reply:
Done